### DISCUSSION TITLE
Integrating with Partytown

### DISCUSSION THREAD

**Discussion by nmenic01** [orestbida/cookieconsent#633] [Q&A]
Hey there,

Is it possible to integrate CookieConsent  with Partytown? Are there any recommended practices or considerations for integrating these two components effectively?

Thank you! :)

**Comment by orestbida** [comment]
The plugin allows the use of a custom type via the `data-type ` attribute, so it should be possible:

```html
<script 
    type="text/plain" 
    data-type="text/partytown" 
    data-category="analytics"
>
    console.log("analytics enabled");
</script>

<script
    type="text/plain"
    src="my-service.js"
    data-category="analytics"
    data-service="My service"
    data-type="text/partytown"
></script>
```

When the plugin is executed, all the script tags that should be enabled (based on user's preferences) will have the `text/partytown` type. You will then need to run partytown after the following events in order to enable the scripts:

- [onConsent](https://cookieconsent.orestbida.com/advanced/callbacks-events.html#onconsent)
- [onChange](https://cookieconsent.orestbida.com/advanced/callbacks-events.html#onchange)
